### PR TITLE
Clippy #... is it 10 already? 

### PR DESCRIPTION
**Pull request by roxy-dao**
Part of #3017

# 👩🏻‍💻 What does this PR do?
Some more clippy fixes... This was getting a bit big... so next PR should be the last and include lint ignores~

## 💌 Any notes for the reviewer?

# 🧪 Testing

- [ ] Make sure tests still run

# 📃 Documentation

- [ ] **Part of an epic**: documentation will be completed for the feature as a whole
- [x] **No documentation required**: no user facing changes or a bug fix which isn't a change in behaviour
- [ ] **These areas should be updated or checked**: 
  1.
  2.

### CODE REVIEW

File: server/service/src/sync/translations/special/name_merge.rs [hunk 1]
```diff
@@ -122,9 +122,9 @@ impl SyncTranslation for NameMergeTranslation {
                     // The remaining NSJ that we keep must logically OR each of these fields with the corresponding field in the deleted NSJs.
                     // We prefer making the name visible to stores rather than losing visibility as it allows users to still make invoices and orders
                     if (!nsj_keep.name_store_join.name_is_customer
-                        && nsj_keep.name_store_join.name_is_customer)
+                        && nsj_delete.name_store_join.name_is_customer)
```
Comment (roxy-dao) [inline server/service/src/sync/translations/special/name_merge.rs:125]: An actual bug... 👀

File: server/android/src/android.rs [hunk 2]
```diff
@@ -23,6 +23,8 @@ pub mod android {
     static SERVER_BUCKET: Mutex<Option<ServerBucket>> = Mutex::new(None);
 
     #[no_mangle]
+    /// # Safety
+    /// TODO: Add why this is unsafe
```
Comment (clemens-msupply) [inline server/android/src/android.rs:27]: Good question, looks like its not needed? have you tried without it? I copied it from an example, maybe it was needed at some earlier version of the jni...

File: server/android/src/android.rs [hunk 3]
```diff
@@ -81,6 +83,7 @@ pub mod android {
     }
 
     #[no_mangle]
+    /// # Safety
```
Comment (clemens-msupply) [inline server/android/src/android.rs:86]: Is this just part of the TODO? or does this comment has a purpose?

File: server/service/src/sync/translations/special/name_merge.rs (hunk 1 above)
Comment (clemens-msupply) [reply server/service/src/sync/translations/special/name_merge.rs:125]: I guess clippy got it? nice!

Comment (roxy-dao) [reply server/service/src/sync/translations/special/name_merge.rs:125]: It was just confused XD

File: server/android/src/android.rs (hunk 3 above)
Comment (roxy-dao) [reply server/android/src/android.rs:86]: Part of above but removed unsafe and seems fine so deleting comments

File: server/graphql/cold_chain/src/types/sensor.rs [hunk 4]
```diff
@@ -130,7 +130,7 @@ impl SensorNode {
     pub async fn assets(&self, ctx: &Context<'_>) -> Result<AssetConnector> {
         let location_id = match &self.row().location_id {
             Some(location_id) => location_id,
-            None => return Ok(AssetConnector::new()),
```
[the remark this answers is not in the document]
Comment (roxy-dao) [reply server/graphql/cold_chain/src/types/sensor.rs:133]: Cool huh~?

**Review by clemens-msupply — APPROVED** [review]
Thanks! Didn't look at all files though...

**Pull request merged**